### PR TITLE
add support for 'devel' log level

### PR DESCRIPTION
**Pull request by boegel**
Currently, debug logs are _very_ verbose, with lots of information being logged that is only really useful when developing EasyBuild features.

One particular example in the logging by the toolchain mechanism.

With this `devel` log level in place, we could avoid including stuff like this in `debug` logs, while avoiding having to remove the `log.debug` statements entirely.

Once a review of all current `log.debug` statements is done and the relevant ones are 'downgraded' to `log.devel`, we can consider make debug logging the default in EasyBuild v3.0.0 (potentially combined with zipping logs by default).

cc @wpoely86, @ocaisa

### CODE REVIEW

**Comment by boegel** [comment]
relates to #1353 and #1796

**Comment by boegel** [comment]
@wpoely86 please review?

The 2nd step here would be to actually switch to using `log.devel` in some places in the framework (e.g. a large part of the `toolchain` mechanism), to make debug logs more lightweight/focused.

File: easybuild/tools/build_log.py [hunk 1]
```diff
@@ -58,6 +59,10 @@
 DRY_RUN_MODULES_INSTALL_DIR = None
 
 
+DEVEL_LOG_LEVEL = logging.DEBUG - 1
```
Comment (wpoely86) [inline easybuild/tools/build_log.py]: not +1 ?

Comment (boegel) [inline easybuild/tools/build_log.py]: no, it's a lower logging level; `log.devel` messages should not be included when using `--debug`, but `log.debug` messages *should* be included when using `--devel` (see also the added test)

Comment (boegel) [inline easybuild/tools/build_log.py]: @wpoely86 so, is this good to go?

**Comment by boegel** [comment]
Thanks for the review @wpoely86!

**Pull request merged**